scss/_fields.scss

```diff
@@ -5,406 +5,547 @@
+      align-items: center;
+      display: flex;
+      height: $outlined-layout-container-height;
+      margin-top: dp.$eight; // to account for the label overlapping the outline
+      position: relative;
+  
+      // represents the 'outline', basically the border around the input
+      &::after {
+        border-width: $outlined-enabled-container-outline-width;
+        border-color: $outlined-enabled-container-outline-color;
+        border-radius: $outlined-enabled-container-shape;
+        border-style: solid;
+        bottom: 0;
+        content: ' ';
+        height: 100%;
+        left: 0;
+        pointer-events: none; 
+        position: absolute;
+        top: 0;
+        user-select: none;
+        width: 100%;
       }
-
-      &.field-text-filled::after {
-        border-color: color.scale(dark.$md-sys-color-on-surface, $alpha: -62%);
+  
+      // leading icon
+      & > .#{$classname}-leading-icon {
+        color: $outlined-enabled-leading-icon-color;
+        font-size: $outlined-enabled-leading-icon-size;
+        line-height: 1;
+        margin-left: $outlined-layout-left-right-padding-with-icon;
+        pointer-events: none;
+        user-select: none;
+
+        & ~ .#{$classname}-input:placeholder-shown:not(:focus) ~ .#{$classname}-label {
+          left: dp.$fifty-two;
+        }
       }
-
-      .field-text-label,
-      .field-text-input,
-      .field-text-icon-leading,
-      .field-text-icon-trailing {
-        color: color.scale(dark.$md-sys-color-on-surface, $alpha: -62%);
+      
+      // input
+      & > .#{$classname}-input {
+        background: none;
+        border-width: 0;
+        caret-color: $outlined-enabled-caret-color;
+        color: $outlined-enabled-input-text-color;
+        font-family: $outlined-enabled-input-text-font;
+        font-size: $outlined-enabled-input-text-size;
+        font-weight: $outlined-enabled-input-text-weight;
+        letter-spacing: $outlined-enabled-input-text-tracking;
+        line-height: $outlined-enabled-input-text-line-height;
+        margin-left: $outlined-layout-left-right-padding-without-icon;
+        margin-right: $outlined-layout-left-right-padding-without-icon;
+        padding: 0;
+        width: 100%;
+    
+        &:focus {
+          outline: none;
+        }
+    
+        // input is empty and not focused
+        &:placeholder-shown:not(:focus) ~ .#{$classname}-label {
+          font-size: $outlined-enabled-label-text-size;
+          line-height: $outlined-enabled-label-text-line-height;
+          padding-left: 0;
+          padding-right: 0;
+          top: dp.$sixteen;
+        }
       }
-    }
-
-    &:focus-within:not(:has(> .field-text-input:disabled)) {
-      &.field-text-filled::after {
-        border-color: dark.$md-sys-color-primary;
+    
+      // label - by default it's styled in 'populated mode' so that browsers not supporting 'advanced' css are ok
+      & > .#{$classname}-label {
+        background-color: white;
```

Comment:
Label background colour should be inherited from the background. Probably best way would be to include background in outlined text field and possibility to change it with general background classes, or perhaps try `background: inherit;` on the whole element.